File: packages/core/test/fixtures/malicious/csharp/binary-formatter/input.cs
// Defanged: BinaryFormatter.Deserialize on attacker-controlled input.
// BinaryFormatter is the canonical .NET deserialization-RCE primitive.
using System.IO;
using System.Runtime.Serialization.Formatters.Binary;

public class Loader
{
    public static object Load(Stream input)
    {
        var fmt = new BinaryFormatter();
        return fmt.Deserialize(input);
    }
}

File: packages/core/test/fixtures/malicious/java/object-input-stream/input.java
// Defanged: ObjectInputStream.readObject on a network-sourced stream.
// The canonical Java deserialization-RCE shape.
import java.io.ObjectInputStream;
import java.net.Socket;

public class Loader {
    public Object load(Socket s) throws Exception {
        ObjectInputStream ois = new ObjectInputStream(s.getInputStream());
        return ois.readObject();
    }
}
